content/en/docs/how-tos/adding-a-new-secret-to-ci.md

```diff
@@ -4,41 +4,129 @@ date: 2020-12-21T10:08:01-04:00
 draft: false
 ---
 
-If a job needs to access sensitive information, we can seal it into a [Kubernetes secret](https://kubernetes.io/docs/concepts/configuration/secret/) and mount it to the pod which runs the job. There are several OpenShift [clusters in CI build farm](/docs/getting-started/useful-links/#clusters). A user can maintain the secrets in a namespace that the user has access to on `api.ci`. The CI `secret-mirroring` tool can be used to mirror the secrets to other namespaces on all clusters in the build farm. 
+Jobs execute as `Pod`s; those that need access to sensitive information will have access to it through mounted Kubernetes
+[`Secrets`](https://kubernetes.io/docs/concepts/configuration/secret/). Secret data is managed self-service by the owners
+of the data. While there are many OpenShift clusters in the [CI build farm](/docs/getting-started/useful-links/#clusters),
+owners of secret data simply need to add their `Secret` to the `api.ci` cluster and automation will sync the data out to
+all environments where jobs execute. 
 
 ## Add A New Secret
 
-Choose a new namespace, and create a pull request to add a folder with the same name in [release/core-services](https://github.com/openshift/release/tree/master/core-services). The folder should contain the manifests for the namespace and the RBACs regarding to the namespace. If the desired namespace exists already, ask the owners of the namespace if it can be shared. After the pull request is merged, the manifests in the folder will be applied automatically to `api.ci`.
+In order to commit secret data to our system, the data will need to be added to the `api.ci` cluster. Management of the
+secret data can either be done manually by one user or in a semi-automated fashion with a group of managing users.
 
-Log into `api.ci` and create the secret in the namespace chosen above.
+In both cases, all users who wish to interact with the system must [log in](/docs/how-tos/use-registries-in-build-farm/#how-do-i-log-in-to-pull-images-that-require-authentication)
+to the `api.ci` cluster.
 
-Use the secret as the source in [the mapping file](https://github.com/openshift/release/tree/master/core-services/secret-mirroring).
-The secret and any modification on it afterwards will be populated to the targeting namespace on [all clusters](/docs/getting-started/useful-links/#clusters) in the build farm.
+### Management By A Single User
```

Comment:
I would prefer that the namespace exists in our release repo in any case.
So `Management By A Group Of Users` is preferred, even if the group has only 1 member.

When the 2nd user found `already-exist` error of creating the namespace, it is hard to find the owner (to ask if it can be shared) if the existing namespace of the manifest is not in release repo.

Reply:
`oc new-project` will record the requester on the `Namespace` API object, it should not be hard to find.

Reply:
```
$ oc --context app.ci new-project for-hongkai
Now using project "for-hongkai" on server "https://api.ci.l2s4.p1.openshiftapps.com:6443".
$ oc --context app.ci get ns for-hongkai -o "jsonpath={.metadata.annotations.openshift\.io/requester}"
stevekuznetsov
```

Reply:
We don't gain anything by making single users go through the process for `openshift/release` and it's explicitly *not* self-service as they require DPTP approvals, which creates a bottle-neck.